client/provide.go

```diff
@@ -88,26 +88,26 @@ func parseProtocol(tp *proto.TransferProtocol) TransferProtocol {
 		}
 		plBytes, err := cbor.Marshal(&pl)
 		if err != nil {
-			return TransferProtocol{}
+			return TransferProtocol{}, err
 		}
 		return TransferProtocol{
 			Codec:   multicodec.TransportGraphsyncFilecoinv1,
 			Payload: plBytes,
-		}
+		}, nil
 	}
-	return TransferProtocol{}
+	return TransferProtocol{}, nil
 }
 
 // ProvideRequest is a message indicating a provider can provide a Key for a given TTL
 type ProvideRequest struct {
 	Key cid.Cid
-	Provider
+	*Provider
 	Timestamp   int64
 	AdvisoryTTL time.Duration
 	Signature   []byte
 }
 
-var provideSchema, _ = ipld.LoadSchemaBytes([]byte(`
+var provideSchema, provideSchemaErr = ipld.LoadSchemaBytes([]byte(`
```

Comment:
Sorry, I think I didn't explain myself well before. Maybe what we can do here, if this schema is something that must be correct to make everything work, is create a method that will be called on init() and panic if the schema throws an error. Then we will catch schema problems on testing (if someone needs to modify the schema string in the future)